<a href="https://colab.research.google.com/github/Sreecr7/data690_Sree/blob/main/Individual_project/IndividualProjectPartb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Individual Project Part_b

In [70]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import plotly.express as px
import plotly.io as pio

In [71]:
pio.templates


Templates configuration
-----------------------
    Default template: 'plotly_dark'
    Available templates:
        ['ggplot2', 'seaborn', 'simple_white', 'plotly',
         'plotly_white', 'plotly_dark', 'presentation', 'xgridoff',
         'ygridoff', 'gridon', 'none']

In [72]:
import plotly

print(plotly.__version__)

5.3.1


In [73]:
!pip install plotly --upgrade


In [74]:
print(plotly.__version__)


5.3.1


In [75]:
pio.templates.default = "plotly_dark" 

In [76]:
df = pd.read_csv('https://raw.githubusercontent.com/Sreecr7/data690_Sree/main/Individual_project/wdi_data%20(5).csv')
df.head(10)

Unnamed: 0  Year      value  ...      Region         Income Group Lending Type
0           0  2010  11.696078  ...  South Asia  Lower middle income         IBRD
1           1  2011  11.863968  ...  South Asia  Lower middle income         IBRD
2           2  2012  12.335278  ...  South Asia  Lower middle income         IBRD
3           3  2013  12.273557  ...  South Asia  Lower middle income         IBRD
4           4  2014  11.687093  ...  South Asia  Lower middle income         IBRD
5           5  2015  11.357799  ...  South Asia  Lower middle income         IBRD
6           6  2016  11.231543  ...  South Asia  Lower middle income         IBRD
7           7  2017  11.113263  ...  South Asia  Lower middle income         IBRD
8           8  2018  12.185249  ...  South Asia  Lower middle income         IBRD
9           9  2019  12.029132  ...  South Asia  Lower middle income         IBRD

[10 rows x 9 columns]

In [77]:
df.drop(columns=["Unnamed: 0"], inplace=True)
df

Year         value  ...         Income Group    Lending Type
0   2010  1.169608e+01  ...  Lower middle income            IBRD
1   2011  1.186397e+01  ...  Lower middle income            IBRD
2   2012  1.233528e+01  ...  Lower middle income            IBRD
3   2013  1.227356e+01  ...  Lower middle income            IBRD
4   2014  1.168709e+01  ...  Lower middle income            IBRD
..   ...           ...  ...                  ...             ...
94  2016  2.719094e+12  ...          High income  Not classified
95  2017  2.901177e+12  ...          High income  Not classified
96  2018  3.119321e+12  ...          High income  Not classified
97  2019  3.105132e+12  ...          High income  Not classified
98  2020  2.808955e+12  ...          High income  Not classified

[99 rows x 8 columns]

In [78]:
df.head(10)

Year      value  ...         Income Group Lending Type
0  2010  11.696078  ...  Lower middle income         IBRD
1  2011  11.863968  ...  Lower middle income         IBRD
2  2012  12.335278  ...  Lower middle income         IBRD
3  2013  12.273557  ...  Lower middle income         IBRD
4  2014  11.687093  ...  Lower middle income         IBRD
5  2015  11.357799  ...  Lower middle income         IBRD
6  2016  11.231543  ...  Lower middle income         IBRD
7  2017  11.113263  ...  Lower middle income         IBRD
8  2018  12.185249  ...  Lower middle income         IBRD
9  2019  12.029132  ...  Lower middle income         IBRD

[10 rows x 8 columns]

In [79]:
df.sample(5)

Year         value  ...         Income Group    Lending Type
90  2012  2.773360e+12  ...          High income  Not classified
28  2016  6.894755e+00  ...          High income  Not classified
94  2016  2.719094e+12  ...          High income  Not classified
44  2010  4.117313e+11  ...          High income  Not classified
72  2016  4.720127e+11  ...  Lower middle income            IBRD

[5 rows x 8 columns]

In [80]:
df.tail(10)

Year         value  ... Income Group    Lending Type
89  2011  2.698073e+12  ...  High income  Not classified
90  2012  2.773360e+12  ...  High income  Not classified
91  2013  2.760063e+12  ...  High income  Not classified
92  2014  2.876410e+12  ...  High income  Not classified
93  2015  2.771010e+12  ...  High income  Not classified
94  2016  2.719094e+12  ...  High income  Not classified
95  2017  2.901177e+12  ...  High income  Not classified
96  2018  3.119321e+12  ...  High income  Not classified
97  2019  3.105132e+12  ...  High income  Not classified
98  2020  2.808955e+12  ...  High income  Not classified

[10 rows x 8 columns]

# Goods exports

In [81]:
df_Export = df.query('indicator == "BX.GSR.MRCH.CD"')
df_Export.head(10)

Year         value  ...         Income Group Lending Type
33  2010  2.309671e+11  ...  Lower middle income         IBRD
34  2011  3.078475e+11  ...  Lower middle income         IBRD
35  2012  2.983206e+11  ...  Lower middle income         IBRD
36  2013  3.191098e+11  ...  Lower middle income         IBRD
37  2014  3.283869e+11  ...  Lower middle income         IBRD
38  2015  2.723527e+11  ...  Lower middle income         IBRD
39  2016  2.686147e+11  ...  Lower middle income         IBRD
40  2017  3.041069e+11  ...  Lower middle income         IBRD
41  2018  3.320869e+11  ...  Lower middle income         IBRD
42  2019  3.312716e+11  ...  Lower middle income         IBRD

[10 rows x 8 columns]

In [82]:
df_pivot = df_Export.pivot(index =['Year','Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot.reset_index().head()

indicator  index  Year  ...    Lending Type BX.GSR.MRCH.CD
0              0  2010  ...  Not classified   4.117313e+11
1              1  2010  ...            IBRD   2.309671e+11
2              2  2010  ...  Not classified   1.290279e+12
3              3  2011  ...  Not classified   4.835275e+11
4              4  2011  ...            IBRD   3.078475e+11

[5 rows x 8 columns]

In [83]:
fig = px.pie(df_pivot, values='BX.GSR.MRCH.CD', names='Country Name', title='Total Population',hover_name='Country Name')
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

In [84]:
df_2010 = df.query("Year == 2010")
df_2010.sample(5)

Year         value  ...         Income Group    Lending Type
88  2010  2.375402e+12  ...          High income  Not classified
33  2010  2.309671e+11  ...  Lower middle income            IBRD
44  2010  4.117313e+11  ...          High income  Not classified
0   2010  1.169608e+01  ...  Lower middle income            IBRD
22  2010  6.793566e+00  ...          High income  Not classified

[5 rows x 8 columns]

In [85]:
df_2018 = df.query("Year == 2018")
df_2018.sample(5)

Year         value  ...         Income Group    Lending Type
8   2018  1.218525e+01  ...  Lower middle income            IBRD
30  2018  6.911074e+00  ...          High income  Not classified
96  2018  3.119321e+12  ...          High income  Not classified
41  2018  3.320869e+11  ...  Lower middle income            IBRD
85  2018  9.164034e+11  ...          High income  Not classified

[5 rows x 8 columns]

In [86]:
df_2018 = df_2018.query("indicator in['BG.GSR.NFSV.GD.ZS','BX.GSR.MRCH.CD','BM.GSR.GNFS.CD']")

df_2018.head(5)

Year         value  ...         Income Group    Lending Type
8   2018  1.218525e+01  ...  Lower middle income            IBRD
19  2018  2.377823e+01  ...          High income  Not classified
30  2018  6.911074e+00  ...          High income  Not classified
41  2018  3.320869e+11  ...  Lower middle income            IBRD
52  2018  4.686051e+11  ...          High income  Not classified

[5 rows x 8 columns]

In [87]:

df_2010 = df_2010.query("indicator in['BG.GSR.NFSV.GD.ZS','BX.GSR.MRCH.CD','BM.GSR.GNFS.CD']")

df_2010.head(5)

Year         value  ...         Income Group    Lending Type
0   2010  1.169608e+01  ...  Lower middle income            IBRD
11  2010  1.934896e+01  ...          High income  Not classified
22  2010  6.793566e+00  ...          High income  Not classified
33  2010  2.309671e+11  ...  Lower middle income            IBRD
44  2010  4.117313e+11  ...          High income  Not classified

[5 rows x 8 columns]

In [88]:
df_pivot = df_2010.pivot(index =['Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot.reset_index().head()

indicator  index Country Code  ... BM.GSR.GNFS.CD BX.GSR.MRCH.CD
0              0          GBR  ...   7.497538e+11   4.117313e+11
1              1          IND  ...   4.390590e+11   2.309671e+11
2              2          USA  ...   2.375402e+12   1.290279e+12

[3 rows x 9 columns]

In [98]:

fig = px.pie(
    df_pivot,
    values='BX.GSR.MRCH.CD', 
    names='Country Name',
    title='TOP 5 Countries by 2010 Goods exports (BoP, current US$)',
    hover_name='Country Name',
    
)
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

##### The United States has the highest percentage of good exports in 2010, at 68 percent, while India has the lowest, at 11.9 percent.as a result, United States The gpd profit from export is reasonable.

#Imports of goods and services 

In [90]:
df_Import = df.query('indicator == "BM.GSR.GNFS.CD"')
df_Import.head(10)

Year         value  ...         Income Group Lending Type
66  2010  4.390590e+11  ...  Lower middle income         IBRD
67  2011  5.530622e+11  ...  Lower middle income         IBRD
68  2012  5.799086e+11  ...  Lower middle income         IBRD
69  2013  5.604085e+11  ...  Lower middle income         IBRD
70  2014  5.535529e+11  ...  Lower middle income         IBRD
71  2015  4.918801e+11  ...  Lower middle income         IBRD
72  2016  4.720127e+11  ...  Lower middle income         IBRD
73  2017  5.616125e+11  ...  Lower middle income         IBRD
74  2018  6.429601e+11  ...  Lower middle income         IBRD
75  2019  6.194848e+11  ...  Lower middle income         IBRD

[10 rows x 8 columns]

In [91]:
df_pivot1 = df_Import.pivot(index =['Year','Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot1.reset_index().head()

indicator  index  Year  ...    Lending Type BM.GSR.GNFS.CD
0              0  2010  ...  Not classified   7.497538e+11
1              1  2010  ...            IBRD   4.390590e+11
2              2  2010  ...  Not classified   2.375402e+12
3              3  2011  ...  Not classified   8.424488e+11
4              4  2011  ...            IBRD   5.530622e+11

[5 rows x 8 columns]

In [99]:
fig = px.line(
    df_pivot1,
    x="Year", 
    y="BM.GSR.GNFS.CD",
    color='Country Name',
    hover_name = "Country Name",
    title='TOP 5 Countries by 2010 Imports of goods and services (BoP, current US$)',
    labels={"BM.GSR.GNFS.CD":"Imports of goods and services (BoP, current US$)"}
    )

fig.show()



##### Between 2010 and 2020, the United States' imports of goods and services fluctuated significantly. As a result, this could be a deciding factor in determining the economy.

# Trade in services 

In [93]:
df_Trade = df.query('indicator == "BG.GSR.NFSV.GD.ZS"')
df_Trade.head(10)

Year      value  ...         Income Group Lending Type
0  2010  11.696078  ...  Lower middle income         IBRD
1  2011  11.863968  ...  Lower middle income         IBRD
2  2012  12.335278  ...  Lower middle income         IBRD
3  2013  12.273557  ...  Lower middle income         IBRD
4  2014  11.687093  ...  Lower middle income         IBRD
5  2015  11.357799  ...  Lower middle income         IBRD
6  2016  11.231543  ...  Lower middle income         IBRD
7  2017  11.113263  ...  Lower middle income         IBRD
8  2018  12.185249  ...  Lower middle income         IBRD
9  2019  12.029132  ...  Lower middle income         IBRD

[10 rows x 8 columns]

In [94]:
df_pivot2 = df_2010.pivot(index =['Year','Country Code','Country Name','Region','Income Group','Lending Type'],columns = 'indicator', values ='value').reset_index()
df_pivot2.reset_index().head()

indicator  index  Year  ... BM.GSR.GNFS.CD BX.GSR.MRCH.CD
0              0  2010  ...   7.497538e+11   4.117313e+11
1              1  2010  ...   4.390590e+11   2.309671e+11
2              2  2010  ...   2.375402e+12   1.290279e+12

[3 rows x 10 columns]

In [97]:
fig = px.bar(
    df_pivot2,
    x="Country Name",
    y="BG.GSR.NFSV.GD.ZS",
    title='TOP 5 Countries by 2010 Trade in services (% of GDP)',
    labels={"BG.GSR.NFSV.GD.ZS":"Trade in services (% of GDP) impact in the year 2010"},
    color="Country Name"
)

fig.show()

##### With respect to Trade in services the United Kingdom ranks No 1 with 19.34 percent of the market, followed by India with 11.69 percent and the United States with 6.79 percent.

# Conclusion

#### I believe that trade in services and trade in goods exports would give us two very different perspectives on the economy and growth. Imports of goods and services into the United States have risen and fallen over time. However, it is consistent for India and the United Kingdom.